provider/cmd/main.go

```diff
@@ -4,18 +4,45 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"os"
+	"syscall"
 
 	"github.com/ironcore-dev/libvirt-provider/provider/cmd/app"
+	originzap "go.uber.org/zap"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const (
+	errTerminatedApp = "app terminated with error"
+)
+
 func main() {
 	ctx := ctrl.SetupSignalHandler()
 
-	if err := app.Command().ExecuteContext(ctx); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+	cmd, ptrLogger := app.Command()
```

Comment:
Instead of returning a logger from the `app.Command` I would rather instantiate a `setupLog` here similar to what the controller-runtime is doing (but without importing `controller-runtime`). As this logger is only used for panic/exit scenarios it is ok to just use it during app startup. 

/ref https://github.com/kubernetes-sigs/controller-runtime/blob/main/examples/crd/main.go#L37

Reply:
you will have problem with inconsistent format. Because i used logger from command which can be configure with flags and you little hide implementation detail because default controller-runtime logger is null logger (dummy logger without printing message). So you can create edge cases when error messages during start won't be printed because logger wasn't initialize properly.

If you want to avoid double reference i can do retype from controller-runtime logr.Sink or we can discuss using our own wrapper library which i sent you before.

Reply:
I get your point, but the setup logger is only a one of thing used during startup. If something goes sour your should fire an alert anyways. Also I have not experienced missing logs for startup failures on our side (e.g. when you set a non existing flag or some other issue occurs). I would really like to keep the start up code clean and avoid a complex pointer setup.

Reply:
And what flushing of buffers?

Reply:
I don't know why we need to do that. Projects like https://github.com/kubeflow/training-operator/blob/master/cmd/training-operator.v1/main.go#L104-L107 which have 100s of contribs and 1k stars don't do that. Not sure why we need this kind of special buffer flushing?

Reply:
we are using zap logging library which specifically required that: https://github.com/uber-go/zap#quick-start

https://pkg.go.dev/go.uber.org/zap#Logger.Sync

if other ignore that, it is their problem and this problem is extremely hard find out because it depends  on multiple factors.

Reply:
you don't need sync, if you are using controller-runtime zap because default output is standard error output https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/pkg/log/zap/zap.go#L173

which isn't buffered by default. But you can have problem, if you use different writer https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap#WriteTo

Reply:
FYI: kubernetes uses flush for klog too https://github.com/kubernetes/component-base/blob/master/logs/logs.go#L175